agent/handlers/v4/task_metadata_handler.go

```diff
@@ -45,9 +45,12 @@ func TaskMetadataHandler(state dockerstate.TaskEngineState, ecsClient api.ECSCli
 			return
 		}
 
+		task, _ := state.TaskByArn(taskArn)
+
 		seelog.Infof("V4 taskMetadata handler: Writing response for task '%s'", taskArn)
 
-		taskResponse, err := NewTaskResponse(taskArn, state, ecsClient, cluster, az, containerInstanceArn, propagateTags)
+		taskResponse, err := NewTaskResponse(taskArn, state, ecsClient, cluster,
+			az, containerInstanceArn, task.ServiceName, propagateTags)
```

Comment:
can we check that task.ServiceName exists outside the generation of the NewTaskResponse?  I see that it's json output is set as `omitempty` but I still want to be assert that the `ServiceName` field exists outside this object creation.

Reply:
Never mind -- The zero string value is `""` for string type so this will always exist.
```
ServiceName string              `json:"ServiceName,omitempty"`
```

Reply:
yeah, we are using empty string to indicate absence of `serviceName`.